src/main/java/org/testng/IReporter2.java

```diff
@@ -0,0 +1,18 @@
+package org.testng;
+
+import org.testng.reporters.XMLReporterConfig;
+import org.testng.xml.XmlSuite;
+
+import java.util.List;
+
+/**
+ * This interface can be implemented by clients to generate a report.  Its method
+ * generateReport() will be invoked after all the suite have run and the parameters
+ * give all the test results that happened during that run.
+ */
+public interface IReporter2 extends ITestNGListener {
+  /**
+   * Generate a report for the given suites with specified attributes
+   */
+  void generateReport(List<XmlSuite> xmlSuites, List<ISuite> suites, XMLReporterConfig config);
```

Comment:
Do you think it is possible to remove config from the method and pass it by injection in the constructor?

Reply:
Why not?

Reply:
As the configuration is related to the reporter instance and not the generation, it will be more logical to have it in the constructor.
But it could be a bit more complicated to implement.